packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -176,12 +175,19 @@ const withDefaultPublicRoutes = (publicRoutes: RouteMatcherParam | undefined) =>
   if (typeof publicRoutes === 'function') {
     return publicRoutes;
   }
+
   const routes = [publicRoutes || ''].flat().filter(Boolean);
-  if (SIGN_IN_URL) {
-    routes.push(matchRoutesStartingWith(SIGN_IN_URL));
+  // TODO: refactor it to use common config file eg SIGN_IN_URL from ./clerkClient
+  // we use process.env for now to support testing
+  const signInUrl = process.env.NEXT_PUBLIC_CLERK_SIGN_IN_URL || '';
```

Comment:
I absolutely agree that we need to create a utility to read the env/config values, however I think it's best if we mock the module for now, even if it requires changing the authMiddleware.test.ts, instead of pulling this logic out of the ./clerkClient file